### DISCUSSION TITLE
Incorrect handling of escape sequence (Unparsed Character Data) in codestarter qute template?

### DISCUSSION THREAD

**Discussion by joes** [quarkusio/quarkus#25948] [Q&A] [answered]
In my codestarter extension I have a qute template in `my-org-camel-integration/runtime/src/main/codestarts/quarkus/my-org-camel-integration-codestart/base/bin/skaffold_build.tpl.qute.sh`.

According to [Qute Reference](https://quarkus.io/guides/qute-reference) I should be able to mark content that should be rendered but not parsed using the sequence `{|` and ends with the sequence `|}`: 

So I was trying it out in order to escape some sections of the shell-script. 

I am attempting to get this output from my qute tpl:

```bash
./mvnw ${args[@]}
```

So I use the escape sequence:

```plain
{|./mvnw ${args[@]}|}
```

But this results in the "broken" output (from the perspective of bash):
```plain
./mvnw ${args[@|}
```

It seems here that the escaping ended at `]}` rather than at `|}`.

Should this considered to be a bug? Should not an escaped section started with `{|` only end at the next `|}` (not `]}`)

[a comment by an automated account is not shown]

**Reply by mkouba** [reply]
> But this results in the "broken" output (from the perspective of bash):

That's because previously unparsed character data started with `{[` and ended with `]}` and we keep this behavior for backwards compatibility. See also https://quarkus.io/guides/qute-reference#basic-building-blocks

> ./mvnw $\{args[@]}

Andy is right, this is the best solution. You only need to escape the start delimiter.

> But it kind of defeats the purpose of the escape sequence I think...

Why?

**Reply by joes** [reply]
> Why?

Badly worded I guess: 

I guess what I am saying is that the escape sequences should match right? It just seems very counter-intuitive that one could mix and match the different types of "delimiters" for unparsed character data. The reason that it was changed from `{[ ... ]}` to `{| ... |}` was to "avoid some common collisions with constructs from other languages" (see https://quarkus.io/guides/qute-reference#basic-building-blocks). But, keeping it backwards compatible in the way you mention defeats the purpose because it is still not possible to avoid the common collisions with constructs from other languages using the new escape sequence.

**Reply by mkouba** [reply]
That's a good point. The new syntax was added in 1.9 so I believe it's the good time to remove the old version.

**Reply by mkouba** [reply]
@joes Would you care to send a PR? :-)

**Comment by ia3andy** [comment, marked as answer]
Try like this: `./mvnw  $\{args[@]}`

**Reply by ia3andy** [reply]
FYI you may find a lot of examples here: https://github.com/quarkusio/quarkus/tree/main/independent-projects/tools/base-codestarts/src/main/resources/codestarts

**Reply by joes** [reply]
> Try like this: `./mvnw $\{args[@|}`

That only results in the unwanted (incorrect) output of:

```
./mvnw ${args[@|}
```

**Reply by joes** [reply]
I'd rather qute did not try to interpret meta sequences in a section that is clearly escaped with `{|` and `|}`. Is it not the intention with this sequence? Or am I misunderstanding?

**Reply by ia3andy** [reply]
Ha sorry I meant `./mvnw $\{args[@]}`

**Reply by joes** [reply]
This works:
```
./mvnw $\{args[@]\}
```

But it kind of defeats the purpose of the escape sequence I think...

**Reply by ia3andy** [reply]
to escape `{...}` you need `\{...}`

**Reply by ia3andy** [reply]
Regarding the escape sequence maybe it needs to be enabled in the QuteEngine for codestarts, if that's the case, then it would break compatibility of the template with previous CLI, we don't want that.
